File: docs/versioned_docs/version-1.9.0/API-Reference/python-examples/api-files/upload-file-v2-2.py
import os

import requests

url = f"{os.getenv('LANGFLOW_URL', '')}/api/v2/files"

headers = {
    "accept": "application/json",
    "x-api-key": f"{os.getenv('LANGFLOW_API_KEY', '')}",
}

files = {
    "file": open(os.getenv("SAMPLE_UPLOAD_FILE", "docs/docs/API-Reference/fixtures/sample-upload.txt"), "rb"),
}

response = requests.request("POST", url, headers=headers, files=files)
response.raise_for_status()

print(response.text)

for _f in files.values():
    if hasattr(_f, "close"):
        _f.close()

File: docs/versioned_docs/version-1.11.0/API-Reference/javascript-examples/api-files/upload-file-v2-2.js
const fs = require("fs");
const path = require("path");

const fixturesDir = path.join(__dirname, "../../fixtures");
const defaultUpload = path.join(fixturesDir, "sample-upload.txt");
const uploadPath = process.env.SAMPLE_UPLOAD_FILE || defaultUpload;
const uploadBuf = fs.readFileSync(uploadPath);
const uploadName = path.basename(uploadPath);

const url = `${process.env.LANGFLOW_URL ?? ""}/api/v2/files`;

const formData = new FormData();
formData.append("file", new Blob([uploadBuf]), uploadName);

const options = {
  method: "POST",
  headers: {
    accept: "application/json",
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: formData,
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));
